util/chunk/chunk.go

```diff
@@ -167,6 +183,17 @@ func (c *Chunk) IsFull() bool {
 	return c.NumRows() >= c.requiredRows
 }
 
+// Prune creates a new Chunk according to `c` and prunes the columns
+// whose index is not in `usedColIdxs`
+func (c *Chunk) Prune(usedColIdxs []int) *Chunk {
+	chk := renewEmpty(c)
```

Comment:
Creating a new `Chunk` whose `columns` reference to the original `Chunk` seems dangerous.
How about letting the original `Chunk` prune itself and returning nothing?
```
func (c *Chunk) Prune(usedColidxs []int)
```

Reply:
Sometimes the original `Chunk` should not be changed. Like in `joiner`, the original `Chunk` might be `baseJoiner.chk` which is reused in entire session and cannot be changed.